demos/python_demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.py

```diff
@@ -15,241 +15,581 @@
+            self.labels_layer = None
+            self.default_label = default_label
+            log.info('Treating detector as a class-agnostic one with output label {}.'.format(self.default_label))
+
+        self.input_size = input_size
+        self.bboxes_layer = self.find_layer_bboxes_output(all_outputs)
+        log.info('Use auto-detected output "{}" as the one containing detected bounding boxes.'
+                 .format(self.bboxes_layer))
+
+    @staticmethod
+    def find_layer_bboxes_output(all_outputs):
+        filter_outputs = [
+            output_name for output_name, out_data in all_outputs.items()
+            if len(np.shape(out_data)) == 2 and np.shape(out_data)[-1] == 5
+        ]
+        if not filter_outputs:
+            raise ValueError('Suitable output with bounding boxes is not found')
+        if len(filter_outputs) > 1:
+            raise ValueError('More than 1 candidate for output with bounding boxes.')
+        return filter_outputs[0]
+
+    @staticmethod
+    def find_layer_by_name(name, all_outputs):
+        return MultipleOutputParser.find_layer(name, all_outputs)
+
+    def __call__(self, outputs):
+        bboxes = outputs[self.bboxes_layer]
+        scores = bboxes[:, 4]
+        bboxes = bboxes[:, :4]
+        bboxes[:, 0::2] /= self.input_size[0]
+        bboxes[:, 1::2] /= self.input_size[1]
+        if self.labels_layer:
+            labels = outputs[self.labels_layer] + 1
+        else:
+            labels = np.full(len(bboxes), self.default_label, dtype=bboxes.dtype)
+        
+        detections = [Detector.Detection(*bbox, score, label) for label, score, bbox in zip(labels, scores, bboxes)]
+        return detections
+
+
+class Modes(Enum):
+    USER_SPECIFIED = 0
+    MIN_LATENCY = 1
+
+
+class ModeInfo:
+    def __init__(self):
+        self.last_start_time = perf_counter()
+        self.last_end_time = None
+        self.frames_count = 0
+        self.latency_sum = 0
+
 
+def put_highlighted_text(frame, message, position, font_face, font_scale, color, thickness):
+    cv2.putText(frame, message, position, font_face, font_scale, (255, 255, 255), thickness + 1) # white border
+    cv2.putText(frame, message, position, font_face, font_scale, color, thickness)
 
-def get_output_postprocessor(net, bboxes='bboxes', labels='labels', scores='scores'):
-    if len(net.outputs) == 1:
-        output_blob = next(iter(net.outputs))
-        return SingleOutputPostprocessor(output_blob)
-    elif len(net.outputs) >= 3:
-        def find_layer(name, all_outputs):
-            suitable_layers = [layer_name for layer_name in all_outputs if name in layer_name]
-            if not suitable_layers:
-                raise ValueError('Suitable layer for "{}" output is not found'.format(name))
 
-            if len(suitable_layers) > 1:
-                raise ValueError('More than 1 layer matched to "{}" output'.format(name))
+def get_plugin_configs(args):
```

Comment:
It's cleaner to pass ` args.num_streams`, `args.device` and `args.number_threads` explicitly as arguments.